go/tools/builders/pack.go

```diff
@@ -354,13 +352,18 @@ func appendFiles(goenv *env, archive string, files []string) error {
 	return goenv.runCommand(args)
 }
 
+type readWithCloser struct {
```

Comment:
you can use https://golang.org/pkg/io/#ReadCloser in the stdlib

Reply:
`io.ReadCloser` is an interface, but I can't create a new instance of it using `io.LimitReader` as the reader and `bufio.Reader` as the closer. This struct implements that interface